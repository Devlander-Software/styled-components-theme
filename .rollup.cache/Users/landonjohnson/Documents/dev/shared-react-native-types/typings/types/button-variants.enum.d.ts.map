{"version":3,"file":"button-variants.enum.d.ts","sourceRoot":"./","sources":["types/button-variants.enum.ts"],"names":[],"mappings":"AAAA,oBAAY,iBAAiB;IACzB,IAAI,SAAS;IACb,QAAQ,aAAa;IACrB,SAAS,cAAc;CAC1B"}