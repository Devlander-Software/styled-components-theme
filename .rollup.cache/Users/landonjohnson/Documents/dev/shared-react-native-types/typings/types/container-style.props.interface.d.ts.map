{"version":3,"file":"container-style.props.interface.d.ts","sourceRoot":"","sources":["../../src/types/container-style.props.interface.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,cAAc,EAAE,MAAM,eAAe,CAAA;AAC9C,OAAO,EAAE,yBAAyB,EAAE,6BAA6B,EAAE,MAAM,gCAAgC,CAAC;AAC1G,OAAO,EAAE,SAAS,EAAE,MAAM,cAAc,CAAC;AACzC,MAAM,WAAW,mBAAoB,SAAQ,SAAS,EAAE,6BAA6B;IACjF,WAAW,CAAC,EAAE,MAAM,CAAC;IACrB,YAAY,CAAC,EAAE,MAAM,CAAC;IACtB,SAAS,CAAC,EAAE,MAAM,CAAC;IACnB,YAAY,CAAC,EAAE,MAAM,CAAC;IACtB,OAAO,CAAC,EAAE,MAAM,CAAC;IACjB,IAAI,CAAC,EAAE,MAAM,CAAC;IAEd,KAAK,CAAC,EAAE,MAAM,CAAC;IACf,UAAU,CAAC,EAAE,MAAM,CAAC;IACpB,UAAU,CAAC,EAAE,MAAM,CAAC;IACpB,WAAW,CAAC,EAAE,MAAM,CAAC;IACrB,aAAa,CAAC,EAAE,MAAM,CAAC;IACvB,MAAM,CAAC,EAAE,MAAM,CAAC;IAChB,QAAQ,CAAC,EAAE,MAAM,CAAC;IAClB,SAAS,CAAC,EAAE,MAAM,CAAC;IAEnB,SAAS,CAAC,EAAE,MAAM,CAAC;IAEnB,MAAM,CAAC,EAAE,MAAM,CAAC;IAChB,WAAW,CAAC,EAAE,MAAM,CAAC;CACxB;AAED,oBAAY,kBAAkB;IAC1B,MAAM,WAAW;IACjB,SAAS,eAAe;IACxB,OAAO,aAAa;IACpB,YAAY,kBAAkB;IAC9B,WAAW,iBAAiB;IAC5B,WAAW,iBAAiB;CAC/B;AAED,oBAAY,cAAc;IACtB,MAAM,WAAW;IACjB,SAAS,eAAe;IACxB,OAAO,aAAa;IACpB,OAAO,YAAY;IACnB,QAAQ,aAAa;CACxB;AAID,MAAM,WAAW,gCAAiC,SAAQ,mBAAmB,EAAE,yBAAyB;IACpG,KAAK,EAAE,cAAc,CAAC;CACzB"}