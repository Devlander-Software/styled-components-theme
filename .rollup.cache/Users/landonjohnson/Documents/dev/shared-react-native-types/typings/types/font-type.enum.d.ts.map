{"version":3,"file":"font-type.enum.d.ts","sourceRoot":"./","sources":["types/font-type.enum.ts"],"names":[],"mappings":"AAAA,oBAAY,YAAY;IACpB,KAAK,UAAU;IACf,KAAK,UAAU;IACf,KAAK,UAAU;IACf,aAAa,kBAAkB;CAClC"}