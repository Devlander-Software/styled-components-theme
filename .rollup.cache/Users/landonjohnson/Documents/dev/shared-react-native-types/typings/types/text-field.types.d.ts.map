{"version":3,"file":"text-field.types.d.ts","sourceRoot":"./","sources":["types/text-field.types.ts"],"names":[],"mappings":"AAAA,oBAAY,kBAAkB;IAC1B,IAAI,SAAS;IACb,KAAK,UAAU;IACf,SAAS,cAAc;IACvB,UAAU,eAAe;CAC1B;AAEH,oBAAY,gBAAgB;IACxB,OAAO,YAAY;IACnB,SAAS,eAAe;IACxB,UAAU,gBAAgB;IAC1B,OAAO,YAAY;IACnB,YAAY,kBAAkB;IAC9B,QAAQ,cAAc;CACvB;AAEH,oBAAY,gBAAgB;IACxB,IAAI,SAAS;IACb,GAAG,QAAQ;IACX,WAAW,gBAAgB;IAC3B,IAAI,SAAS;IACb,aAAa,kBAAkB;CAChC;AACH,oBAAY,mBAAmB;IAC3B,IAAI,SAAS;IACb,GAAG,QAAQ;IACX,WAAW,gBAAgB;IAC3B,mBAAmB,wBAAwB;IAC3C,YAAY,iBAAiB;IAC7B,WAAW,gBAAgB;IAC3B,gBAAgB,qBAAqB;IACrC,YAAY,iBAAiB;IAC7B,UAAU,eAAe;IACzB,iBAAiB,sBAAsB;IACvC,SAAS,cAAc;IACvB,QAAQ,aAAa;IACrB,QAAQ,aAAa;IACrB,UAAU,eAAe;IACzB,IAAI,SAAS;IACb,UAAU,eAAe;IACzB,UAAU,eAAe;IACzB,QAAQ,aAAa;IACrB,gBAAgB,qBAAqB;IACrC,UAAU,eAAe;IACzB,kBAAkB,uBAAuB;IACzC,kBAAkB,uBAAuB;IACzC,WAAW,gBAAgB;IAC3B,eAAe,oBAAoB;IACnC,QAAQ,aAAa;IACrB,WAAW,gBAAgB;IAC3B,WAAW,gBAAgB;CAC5B;AAEH,oBAAY,oBAAoB;IAC5B,eAAe,IAAA;IACf,gBAAgB,IAAA;IAChB,iBAAiB,IAAA;IACjB,gBAAgB,IAAA;IAChB,QAAQ,IAAA;IACR,QAAQ,IAAA;IACR,YAAY,IAAA;IACZ,cAAc,IAAA;IACd,aAAa,IAAA;IACb,WAAW,IAAA;IACX,OAAO,KAAA;IACP,QAAQ,KAAA;IACR,MAAM,KAAA;IACN,aAAa,KAAA;IACb,YAAY,KAAA;IACZ,aAAa,KAAA;IACb,qBAAqB,KAAA;IACrB,aAAa,KAAA;IACb,aAAa,KAAA;IACb,UAAU,KAAA;IACV,cAAc,KAAA;IACd,gBAAgB,KAAA;IAChB,wBAAwB,KAAA;IACxB,yBAAyB,KAAA;IACzB,qCAAqC,KAAA;IACrC,yBAAyB,KAAA;IACzB,uBAAuB,KAAA;IACvB,aAAa,KAAA;IACb,gBAAgB,KAAA;IAChB,SAAS,KAAA;IACT,KAAK,KAAA;IACL,kBAAkB,KAAA;IAClB,cAAc,KAAA;IACd,YAAY,KAAA;IACZ,UAAU,KAAA;IACV,cAAc,KAAA;IACd,KAAK,KAAA;IACL,SAAS,KAAA;CACV"}