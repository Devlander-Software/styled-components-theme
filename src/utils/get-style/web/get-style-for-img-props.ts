import { HtmlImgStyleProps, HtmlImgStylePropsWithTheme } from "../../../types/html/img.web.props";

export const getStyleForImgPropsForWeb = ({
  backgroundColorFromTheme,
  maxWidth,
  borderBottomRightRadius,
  borderColorFromTheme,
  borderTopRightRadius,
  maxHeight,
  height,
  theme,
  width,
  borderRadius,
  borderWidth,
  borderBottomLeftRadius,
  borderTopLeftRadius,
}: HtmlImgStylePropsWithTheme): string =>
  `
        ${
          backgroundColorFromTheme
            ? `background-color: ${theme.handleColorFromTheme(
                backgroundColorFromTheme,
                1,
                theme
              )};`
            : ''
        }

        ${
          borderColorFromTheme
            ? `border-color: ${theme.handleColorFromTheme(
                borderColorFromTheme,
                1,
                theme
              )};`
            : ''
        }
        ${
          borderWidth
            ? `border-width: ${theme.handleUnitProps(borderWidth)};`
            : ''
        }
       
        ${width ? `width: ${theme.handleUnitProps(width)};` : ''}

        ${height ? `height: ${theme.handleUnitProps(height)};` : ''}
        ${width ? `width: ${theme.handleUnitProps(width)};` : ''}
        ${maxWidth ? `max-width: ${theme.handleUnitProps(maxWidth)};` : ''}
        ${maxHeight ? `max-height: ${theme.handleUnitProps(maxHeight)};` : ''}

 border-radius: ${
   borderTopLeftRadius
     ? theme.handleUnitProps(borderTopLeftRadius)
     : theme.handleUnitProps(borderRadius)
 } ${
    borderTopRightRadius
      ? theme.handleUnitProps(borderTopRightRadius)
      : theme.handleUnitProps(borderRadius)
  } ${
    borderBottomRightRadius
      ? theme.handleUnitProps(borderBottomRightRadius)
      : theme.handleUnitProps(borderRadius)
  } ${
    borderBottomLeftRadius
      ? theme.handleUnitProps(borderBottomLeftRadius)
      : theme.handleUnitProps(borderRadius)
  }};



    `
