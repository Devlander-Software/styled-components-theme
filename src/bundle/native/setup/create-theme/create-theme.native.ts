import { adjustColor } from '@devlander/colors';
import { defaultFonts } from '../../../shared/defaults/fonts.defaults';
import { lightColors } from '../../../shared/defaults/light-colors.defaults';

import { NativeTheme } from '../../../shared/types/base-theme-types';
import {
  CreateThemeForNative,
  ThemeConfig,
} from '../../../shared/types/create-theme.types';
import { capFontSize } from '../../../shared/utils/cap-font-size';
import { boxShadowOne } from '../../../shared/utils/handle-box-shadows';
import { unitPropsHandler } from '../../../shared/utils/handle-unit-props';
import { getStyleForGhostPropsForNative } from '../../utils/theme-functions/get-style-for-ghost-props.native';
import { getStyleForImgPropsForNative } from '../../utils/theme-functions/get-style-for-img-props.native';
import { getStyleForTextPropsForNative } from '../../utils/theme-functions/get-style-for-text-props.native';
import { getStyleFromButtonPropsForNative } from '../../utils/theme-functions/get-style-from-button-props.native';
import { getStyleFromPropsNative } from '../../utils/theme-functions/get-style-from-props.native';
import { colorThemeHandlerNative } from '../../utils/theme-functions/handle-color-from-theme.native';
import { handleFontFromThemeForNative } from '../../utils/theme-functions/handle-font-for-theme.native';
import { handleFontSizePropsForNative } from '../../utils/theme-functions/handle-font-size-props.native';

export const createThemeForNative: CreateThemeForNative = (
  config: ThemeConfig
): NativeTheme => {
  const { colorPreferences, fontPreferences, deviceOnTheme } = config;
  const theme: NativeTheme = {
    colors: {
      ...lightColors,
      ...colorPreferences,
    },
    fonts: {
      ...defaultFonts,
      ...fontPreferences,
    },

    elevation: {
      level1: '',
      level2: '',
      level3: '',
      level4: '',
      level5: '',
    },
    paddingOnTheme: {
      paddingForScrollView: 10,
      paddingForScrollViewTwo: 10,
    },
    deviceOnTheme: {
      screenWidth: 0,
      statusBarHeight: 0,
      screenHeight: 0,
      isTablet: false,
      hasNotch: false,
      hasDynamicIsland: false,
      platform: 'ios',
      ...deviceOnTheme,
    },
    fontSizeCap: capFontSize,
    colorThemeHandler: colorThemeHandlerNative,
    fontThemeHandler: handleFontFromThemeForNative,
    fontSizePropsHandler: handleFontSizePropsForNative,
    imageStyleGenerator: getStyleForImgPropsForNative,
    buttonStyleGenerator: getStyleFromButtonPropsForNative,
    ghostStyleGenerator: getStyleForGhostPropsForNative,
    textStyleGenerator: getStyleForTextPropsForNative,
    boxShadowOne: boxShadowOne,
    unitPropsHandler: unitPropsHandler,
    layoutStyleGenerator: getStyleFromPropsNative,
    colorAdjuster: adjustColor,
  };

  return theme;
};
