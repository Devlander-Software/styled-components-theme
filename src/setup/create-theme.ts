import { defaultFonts } from '../defaults/fonts.defaults';
import { lightColors } from '../defaults/light-colors.defaults';
import { ColorsInterface } from "../types/color.types";
import { DeviceOnTheme } from '../types/device-on-theme.type';
import { FontsInterface } from '../types/fonts.interface';
import { ThemeInterface, ThemeInterfaceForHTML } from '../types/theme.types';
import adjustColor from '../utils/adjust-color';
import capFontSize from '../utils/get-style/cap-font-size';
import getStyleForContainerProps from '../utils/get-style/get-style-for-container-props';
import getStyleForGhostProps from '../utils/get-style/get-style-for-ghost-props';
import getStyleForTextProps from '../utils/get-style/get-style-for-text-props';
import getStyleForTouchableOpacityProps from '../utils/get-style/get-style-for-touchable-opacity-props';
import boxShadowOne from '../utils/get-style/handle-box-shadows';
import handleColorFromTheme from '../utils/get-style/handle-color-from-theme';
import handleFontFromTheme from '../utils/get-style/handle-font-for-theme';
import handleFontSizeProps from '../utils/get-style/handle-font-size-props';
import handleSnackbarColor from '../utils/get-style/handle-snackbar-color';
import handleUnitProps from '../utils/get-style/handle-unit-props';
import getStyleForGhostPropsForWeb from '../utils/get-style/web/get-style-for-ghost-props.web';
import { getStyleForImgPropsForWeb } from '../utils/get-style/web/get-style-for-img-props';
import getStyleForParagraphProps from '../utils/get-style/web/get-style-for-paragraph-props';
import handleColorFromThemeWeb from '../utils/get-style/web/handle-color-for-theme.web';
import handleFontFromThemeWeb from '../utils/get-style/web/handle-font-for-theme.web';



export interface ThemeConfig {
    colorPreferences?: Partial<ColorsInterface>
    fontPreferences?: Partial<FontsInterface>
    deviceOnTheme?: Partial<DeviceOnTheme>,
}

export const createThemeForNative = (config: ThemeConfig): ThemeInterface => {
    const { colorPreferences, fontPreferences, deviceOnTheme } = config
    const theme: ThemeInterface = {
        colors: {
            ...lightColors,
            ...colorPreferences,
        },
        fonts: {
            ...defaultFonts,
            ...fontPreferences,
        },
        handleSnackbarColor: handleSnackbarColor,
        handleFontSizeProps: handleFontSizeProps,
        handleUnitProps: handleUnitProps,
        getStyleForTouchableOpacityProps: getStyleForTouchableOpacityProps,
        getStyleForContainerProps: getStyleForContainerProps,
        handleGhost: getStyleForGhostProps,
        getStyleForTextProps: getStyleForTextProps,
        boxShadowOne: boxShadowOne,
        boxShadowTwo: boxShadowOne,
        boxShadowThree: boxShadowOne,
        handleColorFromTheme: handleColorFromTheme,
        handleFontFromTheme: handleFontFromTheme,
        elevation: {
            level1: "",
            level2: "",
            level3: "",
            level4: "",
            level5: "",
        },
        padding: {
            paddingForScrollView: 0,
            paddingForScrollViewTwo: 0
        },
        deviceOnTheme: {
            
            screenWidth: 0,
            statusBarHeight: 0,
            screenHeight: 0,
            isTablet: false,
            hasNotch: false,
            hasDynamicIsland: false,
            platform: 'ios',
            ...deviceOnTheme,
        },
        capFontSize: capFontSize,
        adjustColor: adjustColor,
    }

    return theme
}



export const createThemeForWeb = (config: ThemeConfig): ThemeInterfaceForHTML => {
    const { colorPreferences, fontPreferences, deviceOnTheme } = config
    const theme: ThemeInterfaceForHTML = {
        colors: {
            ...lightColors,
            ...colorPreferences,
        },
        fonts: {
            ...defaultFonts,
            ...fontPreferences,
        },
        handleSnackbarColor: handleSnackbarColor,
        handleFontSizeProps: handleFontSizeProps,
        handleUnitProps: handleUnitProps,
        getStyleForContainerProps: getStyleForContainerProps,
        handleGhost: getStyleForGhostPropsForWeb,
        getStyleForImageProps: getStyleForImgPropsForWeb,
        getStyleForTextProps: getStyleForParagraphProps,
        boxShadowOne: boxShadowOne,
        boxShadowTwo: boxShadowOne,
        boxShadowThree: boxShadowOne,
        handleColorFromTheme: handleColorFromThemeWeb,
        handleFontFromTheme: handleFontFromThemeWeb,
        elevation: {
            level1: "",
            level2: "",
            level3: "",
            level4: "",
            level5: "",
        },
        padding: {
            paddingForScrollView: 0,
            paddingForScrollViewTwo: 0
        },
        deviceOnTheme: {
            screenWidth: 0,
            statusBarHeight: 0,
            screenHeight: 0,
            isTablet: false,
            hasNotch: false,
            hasDynamicIsland: false,
            platform: 'ios',
            ...deviceOnTheme,
          
        },
        capFontSize: capFontSize,
        adjustColor: adjustColor,

    }

    return theme
}


interface CreateThemeConfig extends ThemeConfig {
    framework: "react-native" | "web"
}



const createTheme = (config: CreateThemeConfig): ThemeInterface | ThemeInterfaceForHTML => {
    if(config.framework === "react-native") {
        return createThemeForNative(config)
    } else {
        return createThemeForWeb(config)
    }
 
}


export default createTheme