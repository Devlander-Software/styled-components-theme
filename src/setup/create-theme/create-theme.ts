/* eslint-disable simple-import-sort/imports */
import { ColorsInterface } from "../../types/color.types";
import { DeviceOnTheme } from "../../types/device-on-theme.type";
import { FontsInterface } from "../../types/fonts.interface";
import { ThemeInterface } from "../../types/theme.types";

import { createThemeForNative } from "./create-theme.native";
import { createThemeForWeb } from "./create-theme.web";

export interface ThemeConfig {
  colorPreferences?: Partial<ColorsInterface>;
  fontPreferences?: Partial<FontsInterface>;
  deviceOnTheme?: Partial<DeviceOnTheme>;
}

export interface CreateThemeFunc {
  (config: ThemeConfig): ThemeInterface;
}

export const createTheme: CreateThemeFunc =
  createThemeForNative || createThemeForWeb;

export default createTheme;
