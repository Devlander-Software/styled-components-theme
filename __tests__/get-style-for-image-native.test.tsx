import { createThemeForNative } from '../src/bundle/native/setup/create-theme/create-theme.native';
import { getStyleForImgPropsForNative } from '../src/bundle/native/utils/theme-functions/get-style-for-img-props.native';
import { NativeTheme } from '../src/bundle/shared/types/base-theme-types';
import { ImageStyleWithThemeGeneric } from '../src/bundle/shared/types/img-style-props';

describe('getStyleForImgPropsForNative', () => {
  // Mock theme with handlers
  const mockTheme = createThemeForNative({});
  const normalizeCss = (str: string) => str.replace(/\s/g, '');


  it('returns correct CSS string with provided style properties', () => {
    const imageStyleProps: ImageStyleWithThemeGeneric<NativeTheme,number> = {
      backgroundColorFromTheme: 'blackAlpha50',
      maxWidth: 100,
      borderBottomRightRadius: 10,
      borderColorFromTheme: 'whiteAlpha50',
      borderTopRightRadius: 5,
      maxHeight: 200,
      height: 150,
      theme: mockTheme,
      width: 300,
      borderRadius: 15,
      borderWidth: 2,
      borderBottomLeftRadius: 0,
      borderTopLeftRadius: 0,
    };

    const css = getStyleForImgPropsForNative(imageStyleProps as any);
    const expectedCss = `
        background-color: rgba(0, 0, 0, 0.5);
        border-color: rgba(255, 255, 255, 0.5);
        border-width: 2px;
        width: 300px;
        height: 150px;
        width: 300px;
        max-width: 100px;
        max-height: 200px;
        border-radius: 0px 5px 10px 0px;
    `;

    // Verify if theme handlers were called correctly

    // ...and so on for other properties

    // Verify if css string matches expected output (ignoring whitespace and line breaks)
    expect(normalizeCss(css)).toEqual(normalizeCss(expectedCss));
    expect(css).toMatchSnapshot();

  });

  // Additional tests can be written for other scenarios such as missing properties or different theme handlers' behavior
});
