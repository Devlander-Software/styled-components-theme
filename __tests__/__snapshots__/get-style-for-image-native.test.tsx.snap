// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStyleForImgPropsForNative returns correct CSS string with provided style properties 1`] = `
"
        background-color: rgba(0, 0, 0, 0.5);

        border-color: rgba(255, 255, 255, 0.5);
        border-width: 2px;
       
        width: 300px;

        height: 150px;
        width: 300px;
        max-width: 100px;
        max-height: 200px;

          border-radius: 0px 5px 10px 0px;



    "
`;
